[CI] Benchmark workflow fails, C++14 required to build grpc

The benchmark workflow fails in main:

`Run OpenTelemetry-cpp benchmarks`

```
Run CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ci/do_ci.sh bazel.benchmark
  CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ci/do_ci.sh bazel.benchmark
  mkdir -p benchmarks
  mv api-benchmark_result.json benchmarks
  mv sdk-benchmark_result.json benchmarks
  mv exporters-benchmark_result.json benchmarks
  shell: /usr/bin/bash -e {0}
Unable to find image 'otel/opentelemetry-collector:0.38.0' locally
0.38.0: Pulling from otel/opentelemetry-collector
ac16c79f1[2](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:2)d[3](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:3): Pulling fs layer
831[4](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:4)bda3d[5](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:5)41: Pulling fs layer
ae578[6](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:6)[7](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:7)5ac01: Pulling fs layer
ae57[8](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:9)675ac01: Verifying Checksum
ae578675ac01: Download complete
ac16c7[9](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:10)f12d3: Verifying Checksum
ac16c79f12d3: Download complete
ac16c79f12d3: Pull complete
8314bda3d541: Verifying Checksum
8314bda3d541: Download complete
8314bda3d541: Pull complete
ae578675ac01: Pull complete
Digest: sha256:2dae16d44fdd5c865761f64351ea01795772c39a458240c213ed9cf6a946e876
Status: Downloaded newer image for otel/opentelemetry-collector:0.38.0
e2e7214dda7f6fbe03a6c6ad383aecb653cf5872abe42501ee261920d18e8de8
2023/04/01 16:19:32 Downloading https://releases.bazel.build/4.2.0/release/bazel-4.2.0-linux-x86_64...
Starting local Bazel server and connecting to it...
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 54 packages loaded
    currently loading: ext ... (9 packages)
Loading: 63 packages loaded
Loading: 63 packages loaded
Starting local Bazel server and connecting to it...
Loading: 
Loading: 0 packages loaded
Analyzing: 14 targets (7 packages loaded)
Analyzing: 14 targets (7 packages loaded, 0 targets configured)
Analyzing: 14 targets (24 packages loaded, 12 targets configured)
Analyzing: 14 targets (38 packages loaded, 128 targets configured)
Analyzing: 14 targets (44 packages loaded, 498 targets configured)
Analyzing: 14 targets (72 packages loaded, 1745 targets configured)
Analyzing: 14 targets ([10](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:11)8 packages loaded, 3840 targets configured)
INFO: Analyzed 14 targets (130 packages loaded, 4018 targets configured).
INFO: Found 14 targets...
[0 / 121] [Prepa] BazelWorkspaceStatusAction stable-status.txt
[237 / 286] Compiling sdk/test/trace/sampler_benchmark.cc [for host]; 2s linux-sandbox ... (2 actions running)
[238 / 286] Compiling sdk/test/trace/sampler_benchmark.cc [for host]; 3s linux-sandbox ... (2 actions running)
[239 / 286] Compiling sdk/test/trace/sampler_benchmark.cc [for host]; 5s linux-sandbox ... (2 actions running)
[243 / 286] Compiling sdk/src/trace/batch_span_processor.cc [for host]; 0s linux-sandbox ... (2 actions running)
[246 / 286] Compiling sdk/src/trace/tracer.cc [for host]; 3s linux-sandbox ... (2 actions, 1 running)
[250 / 286] Compiling sdk/src/trace/tracer_context_factory.cc [for host]; 1s linux-sandbox ... (2 actions, 1 running)
[254 / 286] Compiling src/benchmark_register.cc [for host]; 0s linux-sandbox ... (2 actions, 1 running)
[262 / 286] Compiling sdk/src/trace/exporter.cc [for host]; 1s linux-sandbox ... (2 actions, 1 running)
[269 / 286] Compiling sdk/src/trace/span.cc [for host]; 2s linux-sandbox ... (2 actions, 1 running)
[280 / 286] Compiling sdk/src/trace/samplers/parent.cc [for host]; 1s linux-sandbox ... (2 actions, 1 running)
[296 / 315] Compiling sdk/test/metrics/measurements_benchmark.cc [for host]; 2s linux-sandbox ... (2 actions, 1 running)
[301 / 315] Compiling sdk/src/metrics/state/observable_registry.cc [for host]; 2s linux-sandbox ... (2 actions, 1 running)
[306 / 315] Compiling sdk/src/metrics/aggregation/sum_aggregation.cc [for host]; 1s linux-sandbox
[318 / 322] Executing genrule //sdk/test/metrics:measurements_benchmark_result; 1s linux-sandbox ... (2 actions running)
[335 / 531] Executing genrule //sdk/test/common:circular_buffer_benchmark_result; 8s linux-sandbox ... (2 actions running)
[348 / 531] Compiling src/google/protobuf/compiler/java/service.cc [for host]; 1s linux-sandbox ... (2 actions running)
[367 / 531] Compiling src/google/protobuf/compiler/java/map_field_lite.cc [for host]; 1s linux-sandbox ... (2 actions running)
[384 / 531] Compiling src/google/protobuf/util/message_differencer.cc [for host]; 1s linux-sandbox ... (2 actions running)
[410 / 531] Compiling src/google/protobuf/text_format.cc [for host]; 2s linux-sandbox ... (2 actions running)
[435 / 531] Compiling src/google/protobuf/compiler/python/generator.cc [for host]; 1s linux-sandbox ... (2 actions running)
[467 / 531] Compiling src/google/protobuf/descriptor.cc [for host]; 10s linux-sandbox ... (2 actions running)
[517 / 648] Compiling src/google/protobuf/util/internal/default_value_objectwriter.cc [for host]; 0s linux-sandbox ... (2 actions running)
INFO: From Generating Descriptor Set proto_library @com_github_cncf_udpa//xds/service/orca/v3:pkg [for host]:
xds/service/orca/v3/orca.proto:14:1: warning: Import validate/validate.proto is unused.
INFO: From Generating Descriptor Set proto_library @com_github_cncf_udpa//xds/type/v3:pkg [for host]:
xds/type/v3/typed_struct.proto:10:1: warning: Import validate/validate.proto is unused.
[845 / 863] Compiling src/compiler/objective_c_generator.cc [for host]; 1s linux-sandbox ... (2 actions running)
INFO: From Action external/com_github_opentelemetry_proto/opentelemetry/proto/collector/trace/v1/trace_service.grpc.pb.h [for host]:
bazel-out/host/bin/external/com_github_opentelemetry_proto/external/com_github_opentelemetry_proto: warning: directory does not exist.
ERROR: /home/runner/.cache/bazel/e6b8d6759295e14b76bc8cb98b604748/external/com_github_grpc_grpc/BUILD:6746:16: Compiling src/cpp/client/client_callback.cc [for host] failed: (Exit 1): gcc-10 failed: error executing command /usr/bin/gcc-10 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining [11](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:12)0 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from external/com_github_grpc_grpc/src/core/lib/gprpp/time.h:31,
                 from external/com_github_grpc_grpc/src/core/lib/iomgr/exec_ctx.h:36,
                 from external/com_github_grpc_grpc/src/cpp/client/client_callback.cc:26:
external/com_github_grpc_grpc/src/core/lib/gpr/useful.h: In function 'int grpc_core::QsortCompare(const absl::lts_20220623::variant<Alts ...>&, const absl::lts_20220623::variant<Alts ...>&)':
external/com_github_grpc_grpc/src/core/lib/gpr/useful.h:109:17: error: use of 'auto' in lambda parameter declaration only available with '-std=c++[14](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:15)' or '-std=gnu++14'
  109 |       [&](const auto& x) {
      |                 ^~~~
INFO: Elapsed time: 259.796s, Critical Path: [18](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:19).54s
INFO: 587 processes: [39](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:40) internal, 5[48](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4584090057/jobs/8095363842#step:5:49) linux-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
Error: Process completed with exit code 1.
```
It is a little bit confusing that there is two "benchmarks". One of them is called with `ci/do_ci.sh bazel.benchmark` which is failed, another one is called with `./ci/do_ci.sh benchmark` succeed. 

Inside the Bash script, `--host_cxxopt=-std=c++14` is missing in one of them, maybe it could be the reason to fail.

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -25,12 +25,12 @@ jobs:
           key: bazel_benchmark
       - name: setup
         run: |
-          sudo -E ./ci/setup_cmake.sh
-          sudo -E ./ci/setup_ci_environment.sh
+          sudo ./ci/setup_ci_environment.sh
+          sudo ./ci/install_bazelisk.sh
       - name: Run benchmark
         id: run_benchmarks
         run: |
-          CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ci/do_ci.sh bazel.benchmark
+          ./ci/do_ci.sh bazel.benchmark
           mkdir -p benchmarks
           mv api-benchmark_result.json benchmarks
           mv sdk-benchmark_result.json benchmarks

--- a/ci/do_ci.sh
+++ b/ci/do_ci.sh
@@ -25,7 +25,7 @@ function run_benchmarks
 
   [ -z "${BENCHMARK_DIR}" ] && export BENCHMARK_DIR=$HOME/benchmark
   mkdir -p $BENCHMARK_DIR
-  bazel $BAZEL_STARTUP_OPTIONS build --cxxopt=-std=c++14 $BAZEL_OPTIONS_ASYNC -c opt -- \
+  bazel $BAZEL_STARTUP_OPTIONS build --host_cxxopt=-std=c++14 --cxxopt=-std=c++14 $BAZEL_OPTIONS_ASYNC -c opt -- \
     $(bazel query 'attr("tags", "benchmark_result", ...)')
   echo ""
   echo "Benchmark results in $BENCHMARK_DIR:"

